### PR TITLE
Changes to support explicit quota checks

### PR DESCRIPTION
**Pull request by mkjpryor**

### CODE REVIEW

**Review by sjpb — APPROVED** [review]
LGTM

File: environments/.caas/inventory/group_vars/all/prometheus.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 ---
 
-# Set Prometheus storage retention size
-prometheus_storage_retention_size: "{{ metrics_db_maximum_size }}GB"
+# We reserve 10GB of the state volume for cluster state, the rest is for metrics
+prometheus_storage_retention_size: "{{ state_volume_size - 10 }}GB"
```
Comment (JohnGarbutt) [inline environments/.caas/inventory/group_vars/all/prometheus.yml:4]: Could this go negative now? I should stop worrying I suppose.

File: environments/.caas/ui-meta/slurm-infra-fast-volume-type.yml [hunk 2]
```diff
@@ -12,6 +12,24 @@ parameters:
     kind: cloud.ip
     immutable: true
 
+  - name: login_flavor
```
Comment (JohnGarbutt) [inline environments/.caas/ui-meta/slurm-infra-fast-volume-type.yml:15]: Arg... haven't we just removed these :(

File: environments/.caas/inventory/group_vars/all/prometheus.yml (hunk 1 above)
Comment (mkjpryor) [reply environments/.caas/inventory/group_vars/all/prometheus.yml:4]: The ui-meta specified a min of 20.

**Review by JohnGarbutt — APPROVED** [review]

**Pull request merged**